app/models/result_set.rb

```diff
@@ -1,8 +1,8 @@
 class ResultSet
   attr_reader :documents, :total
 
-  def self.get(document_type, params)
-    ResultSetParser.parse(FinderFrontend.get_documents(document_type, params))
+  def self.get(finder, params)
+    ResultSetParser.parse(FinderFrontend.get_documents(finder, params), finder)
```

Comment:
Given that you're passing `finder` into both calls, it might be clearer to break this over multiple lines:

```ruby
ResultSetParser.parse(
  FinderFrontend.get_documents(finder, params),
  finder,
)
```